aruna/api/storage/services/v1/collection_service.proto

```diff
@@ -109,8 +109,101 @@ service CollectionService {
   }
 }
 
+
+// WorkspaceService
+//
+// Service to manage "special" anonymous collections / workspaces 
+service WorkspaceService {
+  // CreateWorkspace
+  //
+  // Status: ALPHA
+  //
+  // A new request to create a personal anonymous workspace
+  rpc CreateWorkspace(CreateWorkspaceRequest)
+      returns (CreateWorkspaceResponse) {
+    option (google.api.http) = {
+      post : "/v1/workspace"
+      body : "*"
+    };
+  }
+
+
+  // DeleteWorkspace
+  //
+  // Status: ALPHA
+  //
+  // Delete a workspace
+  rpc DeleteWorkspace(DeleteWorkspaceRequest)
+      returns (DeleteWorkspaceResponse) {
+    option (google.api.http) = {
+      delete : "/v1/workspace/{workspace_id}"
+      body : "*"
+    };
+  }
+
+  // DeleteWorkspace
+  //
+  // Status: ALPHA
+  //
+  // Claims an anonymous workspace, and transfers the owner to a regular user account.
+  rpc ClaimWorkspace(ClaimWorkspaceRequest)
+      returns (ClaimWorkspaceResponse) {
+    option (google.api.http) = {
+      post : "/v1/workspace/{workspace_id}/claim"
+      body : "*"
+    };
+  }
+
+  // MoveWorkspaceData
+  //
+  // Status: ALPHA
+  //
+  // Claims an anonymous workspace
+  rpc MoveWorkspaceData(MoveWorkspaceDataRequest)
+      returns (MoveWorkspaceDataResponse) {
+    option (google.api.http) = {
+      post : "/v1/workspace/{workspace_id}/move/{collection_id}"
+      body : "*"
+    };
+  }
+}
+
 // Models:
 
+message CreateWorkspaceRequest {
+  string project_name = 1;
```

Comment:
Why not project id?

Reply:
I see that it is way more convenient for users to use the project name instead of the id. I think this will also work as the requirements for project name updates are quite high.